File: app-android/app/src/main/java/com/health2u/domain/repository/AppointmentRepository.kt
package com.health2u.domain.repository

import com.health2u.data.Result
import com.health2u.domain.model.Appointment
import kotlinx.coroutines.flow.Flow

interface AppointmentRepository {
    suspend fun getAppointments(): Result<List<Appointment>>
    suspend fun getAppointmentById(id: String): Result<Appointment>
    suspend fun createAppointment(appointment: Appointment): Result<Appointment>
    suspend fun updateAppointment(appointment: Appointment): Result<Appointment>
    suspend fun deleteAppointment(id: String): Result<Unit>
    fun observeAppointments(): Flow<List<Appointment>>
}

File: app-ios/Sources/Health2u/Domain/Repository/AppointmentRepository.swift
import Foundation

public protocol AppointmentRepository: Sendable {
    func getAppointments() async -> Result<[Appointment], APIError>
    func getAppointment(id: String) async -> Result<Appointment, APIError>
    func createAppointment(_ appointment: Appointment) async -> Result<Appointment, APIError>
    func updateAppointment(_ appointment: Appointment) async -> Result<Appointment, APIError>
    func deleteAppointment(id: String) async -> Result<Void, APIError>
    func observeAppointments() -> AsyncStream<[Appointment]>
}
